### PR TITLE
added yandex.disk support and test file

### LINKED ISSUE #6666: [PullRequest] Adding Yandex.Disk download support
Added test and yandex.disk url resolution

To download model weights hosted from yandex.disk use following snippet to create final url:
```python
from urllib.parse import urlencode

base_url = 'https://cloud-api.yandex.net/v1/disk/public/resources/download?'
public_key = 'https://disk.yandex.ru/d/pWEKt6D3qi3-aw'  # Public link to url

# Encoding url to create final url
final_url = base_url + urlencode(dict(public_key=public_key))
print(final_rul)
```

### PR DESCRIPTION
**Pull request by blacky-i**
Fixes #6666 
/black

### Description
Updated download_url to support internally yandex.disk hosting

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by wyli — APPROVED** [review]

File: tests/test_download_url_yandex.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import os
+import tempfile
+import unittest
+from urllib.error import HTTPError
+
+from monai.apps.utils import download_url
+
+YANDEX_MODEL_URL = (
+    "https://cloud-api.yandex.net/v1/disk/public/resources/download?"
+    "public_key=https%3A%2F%2Fdisk.yandex.ru%2Fd%2Fxs0gzlj2_irgWA"
```
Comment (wyli) [inline tests/test_download_url_yandex.py:23]: this test will run regularly by testing pipelines, do you think this URL and the file will be available for those? @blacky-i

Comment (blacky-i) [reply tests/test_download_url_yandex.py:23]: There are two risks in this url:
1. (API)[https://yandex.ru/dev/disk/api/reference/public.html#download] deprecation (which is very unlikely)
2. This little file is on my personal yandex disk, I will not delete it.

Comment (wyli) [reply tests/test_download_url_yandex.py:23]: ok, it seems when there are concurrent downloadings from multiple jobs, it becomes unavailable, for example:

https://github.com/Project-MONAI/MONAI/actions/runs/5403919298/jobs/9817414158

```
======================================================================
ERROR: test_verify (tests.test_download_url_yandex.TestDownloadUrlYandex)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/MONAI/MONAI/tests/test_download_url_yandex.py", line 34, in test_verify
    download_url(url=YANDEX_MODEL_URL, filepath=os.path.join(tempdir, "model.pt"))
  File "/home/runner/work/MONAI/MONAI/monai/apps/utils.py", line 208, in download_url
    with urlopen(url) as response:
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: NOT FOUND

----------------------------------------------------------------------
```

**Pull request merged**